diff --git a/Mathlib/RingTheory/FiniteType.lean b/Mathlib/RingTheory/FiniteType.lean
index 996693a6d..72fb165aa 100644
--- a/Mathlib/RingTheory/FiniteType.lean
+++ b/Mathlib/RingTheory/FiniteType.lean
@@ -140,6 +140,9 @@ theorem iff_quotient_mvPolynomial :
     use s, MvPolynomial.aeval (↑)
     intro x
-    rw [← Set.mem_range, ← AlgHom.coe_range, ← adjoin_eq_range, SetLike.mem_coe, hs]
-    apply mem_top
+    have hx : x ∈ adjoin R (s : Set S) := by
+      rw [hs]
+      exact mem_top
+    rw [← Set.mem_range, ← AlgHom.coe_range, ← adjoin_eq_range]
+    exact hx
   · rintro ⟨s, f, hsur⟩
     exact .of_surjective f hsur
